### PR TITLE
[ONE-1] Initialize Cargo Library

### PR DESCRIPTION
**Pull request by nicballesteros**
- make window
- make a library not a binary
- make library work
- add box example
- add click register

### CODE REVIEW

**Comment by nicballesteros** [comment]
If you want to try this out, on windows, run `cargo run --features windows --example box`. Chances are that you will have to install some dependencies. This will probably require llvm. Go to the llvm release page on github and install the llvm windows installer and let it run for like 30 mins. You might also need visual studio. Lets hope not because I do not like it and its super big.

File: src/orchestrator/redraw.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    canvas: &mut Canvas,
+    left: u32,
+    top: u32,
+    right: u32,
+    bottom: u32,
+    color: u32,
+) {
+    canvas.save();
+
+    let rect = Rect::new(
```
Comment (sripotturu) [inline src/orchestrator/redraw.rs:16]: f32 to u32?

File: src/platform/windows.rs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    };
+    use skia_safe::{
+        gpu::{
+            gl::FramebufferInfo, BackendRenderTarget, SurfaceOrigin
+        },
+        ColorType,
+        Surface,
+    };
+
+    const WIDTH: f64 = 375.0;
```
Comment (sripotturu) [inline src/platform/windows.rs:23]: f32 to u32?

**Review by Lirons01 — APPROVED** [review]

**Review by sripotturu — APPROVED** [review]
LGTM

**Pull request merged**